index.php

```diff
@@ -213,12 +213,45 @@ function gutenberg_scripts_and_styles( $hook ) {
 		'wp-editor',
 		plugins_url( 'editor/build/style.css', __FILE__ ),
 		array( 'wp-blocks' ),
-		filemtime( plugin_dir_path( __FILE__ ) . 'editor/build/style.css' )
+		gutenberg_asset_cache_version( 'editor/build/style.css' )
 	);
 }
 add_action( 'admin_enqueue_scripts', 'gutenberg_scripts_and_styles' );
 
 /**
+ * Set's a script/style version for cache busting.
+ *
+ * 0.1.0
+ *
+ * @param string $file The file path relative to the plugin's root.
+ *
+ * @return string
+ */
+function gutenberg_asset_cache_version( $file = null ) {
+	$version = false;
+
+	if ( defined( 'WP_DEBUG' ) && WP_DEBUG && $file ) {
+		if ( file_exists( plugin_dir_path( __FILE__ ) . $file ) ) {
+			$version = filemtime( plugin_dir_path( __FILE__ ) . $file );
+		}
+	} else {
+		$version = get_transient( 'gutenberg_version' );
+
+		if ( ! $version && function_exists( 'get_plugin_data' ) ) {
+			$plugin = get_plugin_data( __FILE__, false );
+
+			if ( ! empty( $plugin['Version'] ) ) {
+				$version = $plugin['Version'];
+			}
+
+			set_transient( 'gutenberg_version', $version, DAY_IN_SECONDS );
```

Comment:
With the logic here, if the version is not assigned, we'll perform a `get_transient` and `set_transient` on every request. I think we'll either want to:

- Only `set_transient` if `Version` is not empty (move into condition above)
- Allow transient to be set to an empty string, but strictly check this before calling `get_plugin_data`

My preference is toward the second, which is a matter of changing `! $version` to `false === $version` to reflect we only want the condition to match if `get_transient` returns it's empty state, but not for the empty string.

>If the transient does not exist, does not have a value, or has expired, then get_transient will return `false`.

https://codex.wordpress.org/Function_Reference/get_transient

Practically speaking, we know that the version will be assigned (it already is), but if we were so confident, we'd also not need the `! empty( $plugin['Version'] )` check. So one way or another, we should be consistent.